### PR TITLE
Reorganise test/demo locations

### PR DESCRIPTION
**Pull request by angus-g**
This moves regression-only tests out of the demos directory, ~~and suppresses output when running in CI to prevent blowing up storage on the runners~~. We now have:

- `demos` with documented demos, which are also run as regression tests
- `tests/unit` with unit tests
- `tests/*` for the general regression tests (for CI/Gadi)

### CODE REVIEW

File: demos/Makefile [hunk 1]
```diff
@@ -1,19 +1,14 @@
-cases := base_case 2d_compressible_ALA 2d_compressible_TALA viscoplastic_case 2d_cylindrical analytical_comparisons Drucker-Prager_rheology 3d_spherical 3d_cartesian adjoint optimisation_checkpointing scalar_advection_diffusion multi_material
-long_cases := analytical_comparisons parallel_scaling
+cases := base_case 2d_compressible_ALA 2d_compressible_TALA viscoplastic_case 2d_cylindrical Drucker-Prager_rheology 3d_spherical 3d_cartesian adjoint
```
Comment (drhodrid) [inline demos/Makefile:1]: Is this a list of all demos that will be carried forward? If so, should multi-material examples be here too?

**Review by drhodrid — APPROVED** [review]
Aside from my minor comment, I think this looks good. Pending green tick on testing, happy for it to be merged.

File: demos/Makefile (hunk 1 above)
Comment (angus-g) [reply demos/Makefile:1]: This is all the demos that double as regression tests. The multi-material demos themselves don't currently (in master) get run, except for when building the notebooks, but I guess they actually should to make sure they work!

**Comment by drhodrid** [comment]
Yes - I think we need all of the demos that make it through this transition to be run as regression tests, for consistency.

On 24 Jun 2024, at 10:45 AM, Angus Gibson ***@***.***> wrote:



@angus-g commented on this pull request.

________________________________

In demos/Makefile<https://github.com/g-adopt/g-adopt/pull/86#discussion_r1650226917>:

> @@ -1,19 +1,14 @@
-cases := base_case 2d_compressible_ALA 2d_compressible_TALA viscoplastic_case 2d_cylindrical analytical_comparisons Drucker-Prager_rheology 3d_spherical 3d_cartesian adjoint optimisation_checkpointing scalar_advection_diffusion multi_material
-long_cases := analytical_comparisons parallel_scaling
+cases := base_case 2d_compressible_ALA 2d_compressible_TALA viscoplastic_case 2d_cylindrical Drucker-Prager_rheology 3d_spherical 3d_cartesian adjoint


This is all the demos that double as regression tests. The multi-material demos themselves don't currently (in master) get run, except for when building the notebooks, but I guess they actually should to make sure they work!

**Pull request merged**